tcms/testplans/static/testplans/js/get.js

```diff
@@ -0,0 +1,145 @@
+                                    <li><a href="#">Something Else Here</a></li>
+                                    <li role="separator" class="divider"></li>
+                                    <li><a href="#">Separated Link</a></li>
+                                </ul>
+                            </div>
+                        </div>
+                        <div class="list-view-pf-main-info">TC-${testCase.id}: ${testCase.summary}</div>
+                    </div>
+                    <div class="list-group-item-container container-fluid hidden">
+                        <div class="close">
+                            <span class="pficon pficon-close"></span>
+                        </div>
+                        <div class="row">
+                            <div class="col-md-12">Some test case widgets</div>
+                        </div>
+                    </div>
+                </div>`;
+    };
+
+    var addTestCasesAction = function() {
+        $("#confirmed-testcases input[type='checkbox']").change(function(e) {
+            if ($(this).is(":checked")) {
+                $(this).closest('.list-group-item').addClass("active");
+            } else {
+                $(this).closest('.list-group-item').removeClass("active");
+            }
+        });
+
+        // toggle dropdown menu
+        // $("#confirmed-testcases .list-view-pf-actions").on('show.bs.dropdown', function() {
+        //     var $this = $(this);
+        //     var $dropdown = $this.find('.dropdown');
+        //     var space = $(window).height() - $dropdown[0].getBoundingClientRect().top - $this.find('.dropdown-menu').outerHeight(true);
+        //     $dropdown.toggleClass('dropup', space < 10);
+        // });
+
+        // click the list-view heading then expand a row
+        $("#confirmed-testcases .list-group-item-header").click(function(event) {
+            if (!$(event.target).is("button, a, input, .fa-ellipsis-v")) {
+                $(this).find(".fa-angle-right").toggleClass("fa-angle-down")
+                    .end().parent().toggleClass("list-view-pf-expand-active")
+                    .find(".list-group-item-container").toggleClass("hidden");
+            }
+        });
+
+        // click the close button, hide the expand row and remove the active status
+        $("#confirmed-testcases .list-group-item-container .close").on("click", function() {
+            $(this).parent().addClass("hidden")
+                .parent().removeClass("list-view-pf-expand-active")
+                .find(".fa-angle-right").removeClass("fa-angle-down");
+        });
+    };
+
+    var getTestPlanId = function() {
+        var element = document.getElementById('test_plan_pk');
+        if (element && element.dataset && element.dataset.testplanPk) {
+            return element.dataset.testplanPk;
+        }
+
+        return null;
+    };
+
+    // Return the constructor
+    return {
+        init: Constructor,
+    };
+
+})();
+
+TestPlanModule.init();
```

Comment:
This should live in a `$(document).ready()` function, not self initialize.

```
const TestPlanModule = function() {
  return {
    // api methods
  }
};

$(document).ready(function() {
const testPlanModule = TestPlanModule();
testPlanModule.init();
});
```

note the missing singleton pattern. We may at some point decide to move this module definition on its own but for now leave it in the same file.